### PR TITLE
feat: Decouple propagator instance from propagation algorithm

### PR DESCRIPTION
**Pull request by paulgessinger**
In order to separate the two (and make the python bindings a bit less awkward) we split these up.

### CODE REVIEW

**Comment by asalzburger** [comment]
Hi @paulgessinger - I like this, I think we learn quite a bit also for experiment deployment by doing the python bindings.

File: Examples/Algorithms/Propagation/include/ActsExamples/Propagation/PropagationAlgorithm.hpp [hunk 1]
```diff
@@ -126,7 +121,7 @@ class PropagationAlgorithm : public BareAlgorithm {
   /// Constructor
   /// @param [in] cnf is the configuration struct
   /// @param [in] loglevel is the loggin level
```
Comment (robertlangenberg) [inline Examples/Algorithms/Propagation/include/ActsExamples/Propagation/PropagationAlgorithm.hpp:123]:
```suggestion
  /// @param [in] level is the logging level
```

File: Examples/Algorithms/Propagation/include/ActsExamples/Propagation/PropagatorInterface.hpp [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  PropagationOutput execute(
+      const AlgorithmContext& context, const PropagationAlgorithm::Config& cfg,
+      Acts::LoggerWrapper logger,
+      const Acts::NeutralBoundTrackParameters& startParameters) const override {
+    return executeTest(context, cfg, logger, startParameters);
+  }
+
+ private:
+  /// Templated execute test method for
+  /// charged and netural particles
```
Comment (robertlangenberg) [inline Examples/Algorithms/Propagation/include/ActsExamples/Propagation/PropagatorInterface.hpp:84]:
```suggestion
  /// charged and neutral particles
```

File: Examples/Algorithms/Propagation/include/ActsExamples/Propagation/PropagatorInterface.hpp [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  /// @param [in] context is the contextual data of this event
+  /// @param [in] startParameters the start parameters
+  /// @param [in] pathLength the maximal path length to go
+  template <typename parameters_t>
+  PropagationOutput executeTest(
```
Comment (robertlangenberg) [inline Examples/Algorithms/Propagation/include/ActsExamples/Propagation/PropagatorInterface.hpp:89]: why do you have this private? why not expose the pathLength?

**Review by robertlangenberg — APPROVED** [review]
looks clean, does what you intend to do.

**Pull request merged**